src/ray/raylet/scheduling_queue.cc

```diff
@@ -247,7 +247,8 @@ std::vector<Task> SchedulingQueue::RemoveTasks(std::unordered_set<TaskID> &task_
   return removed_tasks;
 }
 
-Task SchedulingQueue::RemoveTask(const TaskID &task_id, TaskState *removed_task_state) {
+boost::optional<Task> SchedulingQueue::RemoveTask(const TaskID &task_id,
```

Comment:
What do you think about a signature like this instead?
```
bool SchedulingQueue::RemoveTask(const TaskID &task_id, TaskState *removed_task_state, Task *removed_task);
```
I was thinking that it might be good to make sure that the caller doesn't try to use the `TaskState` if the function returns false.

Reply:
I did that first (and it's nicer), but it involved needing a default constructor for Task and TaskSpecification and TaskExecutionSpecification :)